### PR TITLE
Increasing pagination size

### PR DESCRIPTION
**Pull request by seansu4you87**
Bumping to 500 temporarily for Dashboard.  Also removed n+1 query by
removing rolify related code from object initialization hooks.
Performance of GET /incidents increased 2-3x, running around 1.7 secs
locally which isn't great, but we can work on that later

### CODE REVIEW

File: app/controllers/api/v1/incidents_controller.rb [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
-    scope = Incident.search_by( user_id: params[:user_id], 
-                                reactions: params[:reactions],
-                                tags: params[:tags],
-                                ratings: params[:ratings],
-                                incident_type_id: params[:incident_type_id],
-                                place_id: params[:place_id]
-                                )
-    @incidents = scope.page(params[:page]).per(100)
+    scope = Incident
+                .includes(:user)
```
Comment (kidbombay) [inline app/controllers/api/v1/incidents_controller.rb]: You can do these includes in the Incident.search_by function.

File: app/models/user.rb [hunk 2]
```diff
@@ -24,7 +24,9 @@ class User < ApplicationRecord
   
   def setup
     ensure_uid
-    set_default_role
+
+    # WARNING: We should rethink how this works.  This is ripe to cause n + 1 queries
+    set_default_role if id == nil
```
Comment (kidbombay) [inline app/models/user.rb]: This isn't related. I would rather fix it in a diff commit.
Agreed it should be removed.

File: config/initializers/rollbar.rb [hunk 3]
```diff
@@ -12,14 +12,14 @@
   # By default, Rollbar will try to call the `current_user` controller method
   # to fetch the logged-in user object, and then call that object's `id`,
   # `username`, and `email` methods to fetch those properties. To customize:
-  # config.person_method = "my_current_user"
-  # config.person_id_method = "my_id"
-  # config.person_username_method = "my_username"
-  # config.person_email_method = "my_email"
+  # config.person_method = 'my_current_user'
```
Comment (kidbombay) [inline config/initializers/rollbar.rb]: Can we not make unrelated commits?

File: spec/acceptance/incidents_spec.rb [hunk 4]
```diff
@@ -64,14 +64,14 @@
...
+    example "Search results will page with size 500" do
       Incident.transaction do
-        100.times { FactoryGirl.create(:incident) }
+        500.times { FactoryGirl.create(:incident) }
       end
 
       do_request(write_key: api_key.write_key)
       incidents = JSON.parse(response_body)["data"]
-      expect(incidents.count).to eq 100
+      expect(incidents.count).to eq 500
```
Comment (kidbombay) [inline spec/acceptance/incidents_spec.rb]: Still don't agree we should move page size to 500.
There is not a single public API on the web I know that does this. It's just bad form.
Our client apps should account for reasonable API limitations.

**Comment by kidbombay** [comment]
You'll get a much larger performance increase by caching incident api requests.
Since our data is almost always static (incidents update after creation rarely, the cache will almost always hit)

**Comment by kidbombay** [comment]
Let's also look at it from a user side.
Calling 100 objects will always be faster than 500.

The front end can call 100.
Make the dashboard clickable
And then load additional data points as needed.

By moving to page size of 500 which will always be slower
You force all clients to get 500 objects, rarely any of which will be interacted with
And requires full download before interaction can occur.

The metaphor would be in google maps, you only load the tiles you are using. When you move, you load additional tiles. You don't load all tiles ahead of time. It's before client to allow user to interact sooner.

**Comment by seansu4you87** [comment]
@kidbombay agree with long term thoughts on performance.  500 is a very large page size.  However, our only user is the Dashboard, and they don't want to deal with the client side logic to handle pagination.  They're ok with a slightly slower user experience *right now*, for a demo/alpha experience.

**Comment by kidbombay** [comment]
Temporarily I get that. However we all know people rarely go back and fix things.
Do what you need for the demo today.
